libr/socket/socket.c

```diff
@@ -494,12 +494,12 @@ R_API bool r_socket_listen(RSocket *s, const char *port, const char *certfile) {
 		return false;
 	}
 #endif
-
-	switch (s->proto) {
-	case R_SOCKET_PROTO_DEFAULT:
+	if (s->proto == R_SOCKET_PROTO_NONE) {
```

Comment:
just add this to the switch below

Reply:
NONE is not a proto by itself is kind of pre-check before entering the switch, so imho its cleaner to have this early if instead. but i an do the fallback thing if u like

Reply:
Yeah I think I prefer
```C
switch (s->proto) {
case R_SOCKET_PROTO_NONE:
case R_SOCKET_PROTO_TCP:
    s->proto = R_SOCKET_PROTO_TCP;
    .....
}
```
BTW, the `s->proto = R_SOCKET_PROTO_TCP` is not needed if you don't do as I suggested above.

Reply:
Why marked as resolved?

Reply:
your solution is wrong because its asuming that TCP is the DEFAULT, when we already have a DEFAULT variable that can change at any time and shouldnt break the rest of the code. so having NONE as an alias for TCP is wrong. putting NONE in the swith table is also wrong because it forces to refactor the code in the switch table everytime the default protocol is changed. shouldnt happenfrequently, but its an unnecessary weak point. and actually imho it should NEVER be NONE. because the socket constructor must use DEFAULT, and the connect/listen methods are tied to the protocol so that field may be set before reaching that point.